Added content from KB article, fixed titles, broken xrefs. (#319)
mulesoft/docs-runtime-manager: modules/ROOT/pages/lb-cert-validation.adoc

]
----

To get the values for `myVpcID` and `myDlbID` values from the CLI, use the xref:anypoint-platform-cli-commands.adoc#cloudhub-vpc-describe-json[`cloudhub vpc describe-json`] and xref:anypoint-platform-cli-commands.adoc#cloudhub-load-balancer-describe-json[`cloudhub load-balancer describe-json`] commands.

To update any other property, send a `PATCH` request to your load balancer endpoint.

== Manage Certificate Cipher Suites

Using the CloudHub API, you can:

* <<List Available Cipher Suites>>
* <<Display the Default Cipher Suite for a DLB>> 
* <<Rotate the Cipher by Replacing the Default Cipher Suite>>

> 

Reviewer: I dunno'.... If we are introducing this list of links as completions to an intro. phrase, should we change the caps. to sentence style and end with periods?

An alternative is to imply in the intro. text that these are titles: "Using the CloudHub API, you can perform the tasks described in the following texts:"

=== Prerequisites

> To manage cipher suites with the CloudHub API:

Reviewer: Before you can manage...
or
..., you must first:"

* <<Generate Authorization Bearer Token, Generate your authorization bearer token>>.

> * Retrieve your organization ID using the CloudHub API:

Reviewer: ```suggestion
* Retrieve your organization ID using the CloudHub API.
```

+
See https://anypoint.mulesoft.com/exchange/portals/anypoint-platform/f1e97bc6-315a-4490-82a7-23abe036327a.anypoint-platform/cloudhub-api/minor/1.0/console/method/%233736/[Retrieve a single organization].

> * Retrieve your VPC ID using the CloudHub CLI:

Reviewer: ```suggestion
* Retrieve your VPC ID using the CloudHub CLI.
```

+
See xref:anypoint-platform-cli-commands.adoc#cloudhub-vpc-describe-json[cloudhub vpc describe-json].

> * Retrieve your DLB ID using the CloudHub CLI:

Reviewer: ```suggestion
* Retrieve your DLB ID using the CloudHub CLI.
```

+
See xref:anypoint-platform-cli-commands.adoc#cloudhub-load-balancer-describe-json[cloudhub load-balancer describe-json].
* Install `jq` to format the JSON output.
+
See https://stedolan.github.io/jq/[jq].

> === Generate Authorization Bearer Token 

Reviewer: Oh dear. So, this looks like it should be at ==== level...which we do not want to do.
Perhaps we need a Prepare to Manage.... topic to cover all these prereq. tasks, followed by the Manage topic that states that you must have already completed the tasks in Prepare to Manage...?

To generate your authorization bearer token for Anypoint Platform:

> . Send your Anypoint Platform user name and password to CloudHub API:

Reviewer: ```suggestion
. Send your Anypoint Platform user name and password to the CloudHub API:
```

+
[source,json,linenums]
----
curl -X POST 'https://anypoint.mulesoft.com/accounts/login' -H 'Content-Type: application/json' -d '{"username":"myUsername","password":"myPassword"}'
----
+

> This command returns the access token, for example:

Reviewer: ```suggestion
This command returns the access token:
```

+
[source,linenums]
----
{
  "access_token": "myAccessToken",
  "token_type": "bearer",
  "redirectUrl": "/home/"
}
----
. Copy the value of `access_token` to use in the API calls.

=== List Available Cipher Suites 

> To list available cipher suites for an organization using Cloudhub API, run this GET call:

Reviewer: ```suggestion
To list available cipher suites for an organization using the Cloudhub API, run this GET call:
```

[source,linenums]
----
curl https://anypoint.mulesoft.com/cloudhub/api/organizations/myOrgID/loadbalancers/ciphersuites -H 'Authorization: Bearer myAccessToken' | jq
----

* _myOrgID_ is the Organization ID.
* _myAccessToken_ is the access token you generated.

> The output includes `ciphers`, for example:

Reviewer: ```suggestion
The output includes `ciphers`:
```

          "ciphers": "ECDHE-RSA-AES256-GCM-SHA384:ECDHE-RSA-AES128-GCM-SHA256:DHE-RSA-AES256-GCM-SHA384:DHE-RSA-AES128-GCM-SHA256:ECDHE-RSA-AES256-SHA384:ECDHE-RSA-AES128-SHA256:ECDHE-RSA-AES256-SHA:ECDHE-RSA-AES128-SHA:DHE-RSA-AES256-SHA256:DHE-RSA-AES128-SHA256:DHE-RSA-AES256-SHA:DHE-RSA-AES128-SHA:AES256-GCM-SHA384:AES128-GCM-SHA256:AES256-SHA256:AES128-SHA256:AES256-SHA:AES128-SHA:HIGH:!aNULL:!eNULL:!EXPORT:!DES:!MD5:!PSK:!RC4",
          "description": "Default ciphers (old version)",
          "defaultSuite": false
      },
      {
          "id": "5b4bfedee4b0f45ab1de6979",
          "name": "EcdheEcdsaRsa",
          "ciphers": "ECDHE-ECDSA-AES256-GCM-SHA384:ECDHE-ECDSA-AES256-SHA384:ECDHE-RSA-AES256-GCM-SHA384:ECDHE-RSA-AES256-SHA384",
          "description": "ECDHE ECDSA RSA ciphers",
          "defaultSuite": false
      },
      {
          "id": "5ba28106e4b00522d78f40b6",
          "name": "RestrictCiphers",
          "ciphers": "AES256-GCM-SHA384:AES128-GCM-SHA256:AES256-SHA256:AES128-SHA256",
          "description": "Restricted Ciphers",
          "defaultSuite": false
      }
  ],
  "total": 4
}
----

=== Display the Default Cipher Suite for a DLB 

> To display the default cipher suite for a DLB using Cloudhub API, run this GET call:

Reviewer: ```suggestion
To display the default cipher suite for a DLB using the Cloudhub API, run this GET call:
```

[source,linenums]
----
curl https://anypoint.mulesoft.com/cloudhub/api/organizations/myOrgID/vpcs/myVpcID/loadbalancers/myDlbID -H 'Authorization: Bearer myAccessToken' | jq
----

In the command: 

* _myOrgID_ is the Organization ID.
* _myVpcID_ is the ID for the VPC.
* _myDlbID_ is the ID for the DLB.
* _myAccessToken_ is the access token you generated.

> The output includes `defaultSuite`, for example:

Reviewer: ```suggestion
The output includes `defaultSuite`:
```

[source,linenums]
----
"defaultCipherSuite": "ECDHE-RSA-AES256-GCM-SHA384:ECDHE-RSA-AES128-GCM-SHA256:DHE-RSA-AES256-GCM-SHA384:DHE-RSA-AES128-GCM-SHA256:ECDHE-RSA-AES256-SHA384:ECDHE-RSA-AES128-SHA256:ECDHE-RSA-AES256-SHA:ECDHE-RSA-AES128-SHA:DHE-RSA-AES256-SHA256:DHE-RSA-AES128-SHA256:DHE-RSA-AES256-SHA:DHE-RSA-AES128-SHA:AES256-GCM-SHA384:AES128-GCM-SHA256:AES256-SHA256:AES128-SHA256:AES256-SHA:AES128-SHA:HIGH:!aNULL:!eNULL:!EXPORT:!DES:!MD5:!PSK:!RC4"
----

=== Rotate the Cipher by Replacing the Default Cipher Suite

> To rotate the cipher, replace the default cipher suite for a DLB using Cloudhub API, run this GET call:

Reviewer: ```suggestion
To rotate the cipher by replacing the default cipher suite for a DLB using the Cloudhub API, run this GET call:
```

[source,json,linenums]
----
curl -X PATCH https://anypoint.mulesoft.com/cloudhub/api/organizations/myOrgID/vpcs/myVpcID/loadbalancers/myDlbID  -H 'Authorization: Bearer myAccessToken' -H 'Content-Type: application/json' -d '[{"op":"replace", "path":"/defaultCipherSuiteName", "value":"NewDefault"}]' | jq
----

In the command: 

> * _myOrgID_ is the Organization ID.

Reviewer: I should have asked this earlier: do we really need to capitalize Organization in this repeated list item?

* _myVpcID_ is the ID for the VPC.
* _myDlbID_ is the ID for the DLB.
* _myAccessToken_ is the access token you generated.

> * `NewDefault` is the value of the `name` for a cipher suite from the list of available cipher suites that you generated in <<List Available Cipher Suites>>.

Reviewer: ```suggestion
* `NewDefault` is the value of `name` for a cipher suite from the list of available cipher suites that you generated in <<List Available Cipher Suites>>.
```

> The output now includes `defaultCipherSuite`, for example:

Reviewer: ```suggestion
The output now includes `defaultCipherSuite`:
```

[source,linenums]
----
"defaultCipherSuite": "ECDHE-RSA-AES256-GCM-SHA512:DHE-RSA-AES256-GCM-SHA512:ECDHE-RSA-AES256-GCM-SHA384:DHE-RSA-AES256-GCM-SHA384:ECDHE-RSA-AES256-SHA384"
----

=== Recommended Cipher Suites 

> The following recommended ciphers suites provide a good balance between compatibility and security for your SSL endpoint:

Reviewer: ```suggestion
The following recommended cipher suites provide a good balance between compatibility and security for your SSL endpoint:
```

AES128-SHA
----

Most cipher suites offer forward secrecy. RC4-SHA (supports Internet Explorer 8) does not.
For this reason, MuleSoft and Microsoft do not recommend using it.

The CloudHub dedicated load balancer supports TLSv1.1 and TLSv1.2.

> Although you can configure TLS v1.0, PCI compliance doesn't accept this protocol because of its significant vulnerabilities.

Reviewer: ```suggestion
Although you can configure TLSv1.0, PCI compliance doesn't accept this protocol because of its significant vulnerabilities.
```
